fix: CLI command text review (#64)
salesforcecli/plugin-lightning-dev: messages/lightning.preview.app.md

# summary

Preview a Lightning Experience application locally and in real-time, without deploying it.

# description

> In developer preview mode, see local changes to your app in a real-time preview that you don't have to deploy or manually refresh. To let you quickly iterate on your Lightning web components (LWCs) and pages, your app preview automatically refreshes when source code changes are detected.

Reviewer: Ah, I think that's a good call-out. +1 for swapping "developer preview mode" with "this command" or "With Lightning Preview, you can..." or something similar.

When you edit local files in developer preview mode, your LWCs reflect these changes within your {org name} org:

- Basic HTML and CSS edits
- Importing new CSS-only LWCs
- JS edits in-service component library
- JS method changes that don't affect the LWC's public API

> To see any other local changes in your app, you have to deploy to your org. If you make changes directly in your org, (like saving new component properties,) they're immediately live on your app. To update your local version of the app with those changes, you have to retrieve them from your org.

Reviewer: Disclaimer: I may be overthinking this.

I feel like the two separate ideas conveyed in this paragraph need some separation, for clarity and distinction. My proposal is a bulleted approach that's prefaced with a "limitations" statement:

> There are currently a couple of limitations to consider:
> 1. To see any local changes in your app that are not listed above, you have to deploy to your org. [+1 for @jshackell-sfdc 's call-outs to the CLI commands for this, in the comment above.]
> 2. If you make changes directly in your org, (like saving new component properties), they're immediately live on your app. To update your local version of the app with those changes, you have to retrieve them from your org.

I'm concerned that, if these aren't called out more distinctly, both of the messages are going to get lost or muddled.

Author: @khawkins are the names "Lightning Preview" and "developer preview mode" interchangeable? Or is there one "official" name that I should use for consistency?

Reviewer: @moorejacqueline It's a bit of a guess for me as to why the original copy said "developer preview mode". I think it may have been calling out that you have to have your org set to `DEVPREVIEW` mode in order to use this feature (among other qualifications), but that's not worth calling out in this description—which is why I was with Juliet on making it more meaningful.

Alas, I don't think there is as yet an official name for this. I believe it's in the works.

> Use the appropriate topic to preview specific aspects of the development environment.

Author: I'm just going to remove this -- I think it was part of the original CLI text, and I think it refers to uisng `app` vs `site` (vs `component`)

# summary

Preview a Lightning Experience app locally and in real-time, without deploying it.

# description

Enable Local Dev to see local changes to your app in a real-time preview that you don't have to deploy or manually refresh. To let you quickly iterate on your Lightning web components (LWCs) and pages, your app preview automatically refreshes when Local Dev detects source code changes.

> When you edit local files with Local Dev enabled, your org automatically reflects these changes.

Reviewer 2: ```suggestion
When you edit these local files with Local Dev enabled, your org automatically reflects these changes.
```

Reviewer 2: Maybe?  I like it, but up to you.

- Basic HTML and CSS edits to LWCs
- JavaScript changes to LWCs that don't affect the component's public API
- Importing new custom LWCs
- Importing another instance of an existing LWC

To apply any other local changes not listed above, you must deploy them to your org using the `sf project deploy start` command.

> When you make changes directly in your org, (like saving new component properties,) they're automatically deployed to your live app. To update your local version of the app with those changes, you have to retrieve them from your org using the `sf project retrieve start` command.

Reviewer 2: ```suggestion
When you make changes directly in your org (like saving new component properties), they're automatically deployed to your live app. To update your local version of the app with those changes, you must retrieve them from your org using the `sf project retrieve start` command.
```

# flags.device-type.summary

> Type of device to emulate in the preview.

Reviewer: Independent of your change here, we had some discussion about changing the "emulate" verbiage because it was a bit misleading. This is what we came up with. Does this work for you?

```suggestion
Type of device to be used for previewing the content.
```

Author: @khawkins Would something like this be equivalent? 
```suggestion
Type of device to load the app preview.
```

I'm not sure if "load" is an OK verb -- maybe "render" or "run?" We could also do "use," like "Type of device to use for the app preview."

Reviewer: I think "load" could work. Maybe "display" could work as well? I'm good to stick with "load", until someone tells me otherwise.

Author: I like display!

# flags.device-id.summary

> ID of the mobile device to display the preview if device type is set to `mobile`. The default value is the ID of the first available mobile device.

Reviewer 2: right?  These are the possible values for this flag: `<options: desktop|ios|android>`.

> ID of the mobile device to display the preview if device type is set to `mobile`. The default value is the ID of the first available mobile device.

Author: Yes, thanks for catching that!

> ID of the mobile device to emulate the preview when device type is set to `mobile`. The default value is the ID of the first available mobile device.

Reviewer: This is another instance where I'd like to get away from the language of "emulating", because it implies that the preview itself is the emulation, and that's a somewhat incongruent description of what we're doing. I'll make a suggestion that moves away from the language of emulation, but feel free to wordsmith this verbiage further as you will.

```suggestion
ID of the virtual mobile device to be used for previewing, when the device type is set to `mobile`. The default value is the ID of the first available mobile device.
```

# certificate.installation.description

> Before proceeding, make sure you installed the self-signed certificate on your device. The certificate file is located at

Reviewer 3: FYI there was a suggestion by UX to update the flow on mobile and so this particular message may need to be updated/deleted. Will ping you when that is done so that you can take another pass on the newly added/updated strings. But for now we can go with this.

`%s`

To install the certificate, follow these steps:

%s

# certificate.installation.steps.ios

1. Drag and drop the file onto your booted simulator.

> 2. Click `Allow` to proceed with downloading the configuration file.

Reviewer: I don't know what the standard documenting convention is for mobile gestures like this, but the actual action will be a tap, not a click. But maybe users put more emphasis on the repeated designator in documentation, across device experiences? I don't know, so I figured I would ask.

Reviewer: Oh but this will be in a simulator so it doesn't even apply. 🤦 Disregard!

Navigate to `Settings > Security & Location > Advanced > Encryption & credentials` and click `Install from SD card` under `Credential storage`.

# certificate.installation.steps.android.nav-target-api-29

Navigate to `Settings > Security > Encryption & credentials` and click `Install from SD card` under `Credential storage`.

# certificate.installation.steps.android.nav-target-api-30-32

Navigate to `Settings > Security > Encryption & credentials` and click `Install a certificate` under `Credential storage`. Click `CA certificate`, and then click `Install anyway`.

# certificate.installation.steps.android.nav-target-api-33

Navigate to `Settings > Security > More security settings > Encryption & credentials` and click `Install a certificate` under `Credential storage`. Click `CA certificate`, and then click `Install anyway`.

# certificate.installation.steps.android.nav-target-api-34-up

Navigate to `Settings > Security & Privacy > More security & privacy > Encryption & credentials` and click `Install a certificate` under `Credential storage`. Click `CA certificate`, and then click `Install anyway`.

# certificate.waiting

> After you install certificate, press any key to continue...

Reviewer 3: Doesn't read nicely without a `the`

# mobileapp.notfound

%s isn't installed on your device.

# mobileapp.download

%s isn't installed on your device. Do you want to download and install it?

# examples

- Preview the target org "myOrg" in a desktop environment.
  <%= config.bin %> <%= command.id %> --target-org myOrg
- Preview the application "myApp" for the target org "myOrg" in a desktop environment:
  <%= config.bin %> <%= command.id %> --name MyApp --target-org myOrg --device-type desktop
- Preview the target org "myOrg" on an iOS device:

>   <%= config.bin %> <%= command.id %> --target-org myOrg --device-type ios --device-id "37C98F87-63C4"

Reviewer: @maliroteh-sf Should our `--device-id` value in this case be one of the friendly names of the devices, which we also allow for? E.g. `--device-id "iPhone 15 Pro Max"`? Users are simply going to run away screaming from the example given. 😬

Reviewer 3: Oh yes that's a good point. We should switch to using the friendly name in the examples instead, since our code works with both friendly name and ID.

Reviewer: ```suggestion
  <%= config.bin %> <%= command.id %> --target-org myOrg --device-type ios --device-id "iPhone 15 Pro Max"
```

Navigate to `Settings > Security > Encryption & credentials` and click `Install from SD card` under `Credential storage`.

# certificate.installation.steps.android.nav-target-api-30-32

Navigate to `Settings > Security > Encryption & credentials` and click `Install a certificate` under `Credential storage`. Click `CA certificate`, and then click `Install anyway`.

# certificate.installation.steps.android.nav-target-api-33

Navigate to `Settings > Security > More security settings > Encryption & credentials` and click `Install a certificate` under `Credential storage`. Click `CA certificate`, and then click `Install anyway`.

# certificate.installation.steps.android.nav-target-api-34-up

Navigate to `Settings > Security & Privacy > More security & privacy > Encryption & credentials` and click `Install a certificate` under `Credential storage`. Click `CA certificate`, and then click `Install anyway`.

# certificate.waiting

After you install the certificate, press any key to continue...

# mobileapp.notfound

%s isn't installed on your device.

# mobileapp.download

%s isn't installed on your device. Do you want to download and install it?

# examples

> - Preview the target org "myOrg" in a desktop environment:

Reviewer 2: Hmm, not sure what it means to "Preview an org".  I thought this command was for previewing an _app_.  Question: if you don't specify the app name with `--name`, which app does the preview show?  Is there a "default" app?  Or maybe the "first" one (whatever that means)?  Again, my questions show my lack of knowledge!  But if possible, try to rewrite this so you're talking about previewing _apps_ and not orgs.  For example:

Preview the default app in the target org "myOrg" in a desktop environment:

Reviewer: - I think "target org" is a holdover from when `preview app` was `preview org`.
- The "default" app is just whatever app is currently configured to show up when you login. Basically, without `--name`, we don't qualify the launch, and just let the browser experience resolve as it will.

  <%= config.bin %> <%= command.id %> --target-org myOrg

> - Preview the application "myApp" for the target org "myOrg" in a desktop environment:

Reviewer 2: FYI, I noticed you use both "app" and "application" in the help.  Is that okay?  Just checking :)

  <%= config.bin %> <%= command.id %> --name MyApp --target-org myOrg --device-type desktop

> - Preview the target org "myOrg" on an iOS device:

Reviewer 2: Ditto above comment -- try to rewrite so you're explaining the app that's being previewed.
